rama-cli/src/cmd/stunnel/mod.rs

```diff
@@ -0,0 +1,250 @@
+        socket::Interface,
+        tls::{
+            DataEncoding,
+            client::ServerVerifyMode,
+            server::{ServerAuth, ServerAuthData, ServerConfig},
+        },
+    },
+    tcp::{
+        client::service::{Forwarder, TcpConnector},
+        server::TcpListener,
+    },
+    telemetry::tracing::{self, level_filters::LevelFilter},
+    tls::boring::{
+        client::{TlsConnectorDataBuilder, TlsConnectorLayer},
+        core::x509::{X509, store::X509StoreBuilder},
+        server::{TlsAcceptorData, TlsAcceptorLayer},
+    },
+    utils::str::NonEmptyString,
+};
+
+use clap::{Args, Subcommand};
+use std::{path::PathBuf, sync::Arc, time::Duration};
+
+use crate::utils::tls::new_server_config;
+
+#[derive(Debug, Args)]
+/// rama stunnel service
+pub struct StunnelCommand {
+    #[command(subcommand)]
+    pub commands: StunnelSubcommand,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum StunnelSubcommand {
+    Server(ServerArgs),
+    Client(ClientArgs),
+}
+
+#[derive(Debug, Args)]
+pub struct ServerArgs {
+    #[arg(long, default_value = "127.0.0.1:8002")]
+    /// address and port to listen on for incoming TLS connections
+    pub listen: Interface,
+
+    #[arg(long, default_value = "127.0.0.1:8001")]
+    /// backend address to forward decrypted connections to
+    pub forward: SocketAddress,
+
+    #[arg(long, requires = "key")]
+    /// path to TLS certificate file (PEM format)
+    ///
+    /// if not provided, a self-signed certificate will be auto-generated.
+    /// can also be set via RAMA_TLS_CRT environment variable (base64-encoded).
+    pub cert: Option<PathBuf>,
+
+    #[arg(long, requires = "cert")]
+    /// path to TLS private key file (PEM format)
+    ///
+    /// if not provided, a private key will be auto-generated.
+    /// can also be set via RAMA_TLS_KEY environment variable (base64-encoded).
+    pub key: Option<PathBuf>,
+}
+
+#[derive(Debug, Args)]
+pub struct ClientArgs {
+    #[arg(long, default_value = "127.0.0.1:8003")]
+    /// address and port to listen on
+    pub listen: Interface,
+
+    #[arg(long, default_value = "127.0.0.1:8002", value_name = "HOST:PORT")]
+    /// server to connect to (port is REQUIRED)
+    ///
+    /// examples:
+    ///   localhost:8443
+    ///   example.com:8443
+    ///   192.168.1.100:8443
+    ///
+    /// port must be explicitly specified (e.g., :8443, :443)
+    pub connect: Authority,
+
+    #[arg(long, conflicts_with = "insecure")]
+    /// path to CA certificate bundle for server verification (PEM format)
+    ///
+    /// use this to trust a specific CA or self-signed certificate.
+    /// if not provided, the system trust store will be used.
+    pub cacert: Option<PathBuf>,
+
+    #[arg(short = 'k', long)]
+    /// skip TLS certificate verification (INSECURE - testing only!)
+    ///
+    /// this disables all certificate validation and should NEVER be used
+    /// in production. Use --cacert instead for self-signed certificates.
```

Comment:
or ensure that your system keychain has the cert or one of its root certs as a trusted cert.

(^ add something like that to it)